spec/unit/forge/module_release_spec.rb

```diff
@@ -9,33 +9,57 @@
   let(:forge_release_class) { PuppetForge::V3::Release }
 
   let(:download_path) { instance_double('Pathname') }
+  let(:tarball_cache_path) { instance_double('Pathname') }
   let(:unpack_path) { instance_double('Pathname') }
   let(:target_dir) { instance_double('Pathname') }
   let(:file_lists) { {:valid=>['valid_ex'], :invalid=>[], :symlinks=>['symlink_ex']} }
 
   before do
     subject.download_path = download_path
+    subject.tarball_cache_path = tarball_cache_path
     subject.unpack_path = unpack_path
   end
 
-  describe '#download' do
-    it "downloads the module from the forge into `download_path`" do
-      expect(subject.forge_release).to receive(:download).with(download_path)
-      subject.download
+  context "no cached tarball" do
+    describe '#download' do
+      it "downloads the module from the forge into `download_path`" do
+        expect(tarball_cache_path).to receive(:exist?).and_return(false)
+        expect(subject.forge_release).to receive(:download).with(download_path)
+        allow(FileUtils).to receive(:mkdir_p).with(anything())
```

Comment:
Will use instance_variable_set to set that, otherwise it's controlled by the presence/absence of r10k.yaml.